realtime/websocket/processors.py

```diff
@@ -0,0 +1,118 @@
+import asyncio
+import base64
+import io
+import logging
+import time
+import wave
+
+import av
+import numpy as np
+from fastapi import WebSocket
+
+from realtime.streams import AudioStream, ByteStream, TextStream, VideoStream
+
+
+class WebsocketInputStream:
+    """
+    Handles incoming WebSocket messages and streams audio and text data.
+
+    Attributes:
+        ws (WebSocket): The WebSocket connection.
+    """
+
+    def __init__(self, ws: WebSocket):
+        self.ws = ws
+
+    async def run(self, audio_stream: AudioStream, message_stream: TextStream, video_stream: VideoStream):
+        """
+        Starts the task to process incoming WebSocket messages.
+
+        Returns:
+            Tuple[AudioStream, TextStream]: A tuple containing the audio and message streams.
+        """
+        self.audio_output_stream = audio_stream
+        self.message_stream = message_stream
+        self.video_stream = video_stream
+
+        audio_data = b""
+        while True:
+            try:
+                data = await self.ws.receive_json()
+                if data.get("type") == "message":
+                    await self.message_stream.put(data.get("data"))
+                elif data.get("type") == "audio":
+                    audio_bytes = base64.b64decode(data.get("data"))
+                    audio_data += audio_bytes
+
+                    if len(audio_data) < 2:
+                        continue
+                    if len(audio_data) % 2 != 0:
+                        # TODO: Get audio dtype from the frontend instead of hardcoding it
+                        array = np.frombuffer(audio_data[:-1], dtype=np.int16).reshape(1, -1)
+                        audio_data = audio_data[-1:]
+                    else:
+                        # TODO: Get audio dtype from the frontend instead of hardcoding it
+                        array = np.frombuffer(audio_data, dtype=np.int16).reshape(1, -1)
+                        audio_data = b""
+
+                    frame = av.AudioFrame.from_ndarray(array, format="s16", layout="mono")
+                    frame.sample_rate = 8000
```

Comment:
Why is incoming sample rate 8000? and outgoing 16000?

Reply:
fixed.